types/node/globals.d.ts

```diff
@@ -132,25 +134,19 @@ declare class Buffer extends Uint8Array {
      *
      * @param arrayBuffer The .buffer property of any TypedArray or a new ArrayBuffer()
      */
-    static from(arrayBuffer: ArrayBuffer | SharedArrayBuffer, byteOffset?: number, length?: number): Buffer;
+    static from(arrayBuffer: WithImplicitCoercion<ArrayBuffer | SharedArrayBuffer>, byteOffset?: number, length?: number): Buffer;
     /**
      * Creates a new Buffer using the passed {data}
      * @param data data to create a new Buffer
      */
-    static from(data: ReadonlyArray<number>): Buffer;
-    static from(data: Uint8Array): Buffer;
-    /**
-     * Creates a new buffer containing the coerced value of an object
-     * A `TypeError` will be thrown if {obj} has not mentioned methods or is not of other type appropriate for `Buffer.from()` variants.
-     * @param obj An object supporting `Symbol.toPrimitive` or `valueOf()`.
-     */
-    static from(obj: { valueOf(): string | object } | { [Symbol.toPrimitive](hint: 'string'): string }, byteOffset?: number, length?: number): Buffer;
+    static from(data: Uint8Array | ReadonlyArray<number>): Buffer;
+    static from(data: WithImplicitCoercion<Uint8Array | ReadonlyArray<number> | string>): Buffer;
```

Comment:
I combined the two original overloads into one.  In theory I could also combine my two overloads into one, but doing so breaks compatibility with the `Uint8Array` type that `Buffer` extends.

Also `| string` isn't strictly necessary here (covered in the next signature) but including it here seemed pretty harmless, and addresses #38272 where someone tried to pass a union of `string | Buffer` to `Buffer.from`.

Reply:
Do you need `static from(data: Uint8Array...`? It seems like it would be covered by the `static from(data: WithImplicit...` overload.

Reply:
Yeah, that's what I meant in my above comment:

> In theory I could also combine my two overloads into one, but doing so breaks compatibility with the Uint8Array type that Buffer extends.

I do think the `WithImplicit` overload covers everything in the `Uint8Array` overload, but if I get rid of the `Uint8Array` overload, TS errors on the `class Buffer extends Uint8Array` line, saying that the `Buffer` type no longer correctly extends the `Uint8Array` class.  I'm not entirely sure why that error happens, but I couldn't find a better fix than just leaving the theoretically-unnecessary overload.

Reply:
Ah, I see. It needs to exist for compatibility with the static `from` method on `Uint8Array`.